apps/entry/mutations.py

```
@@ -394,42 +378,51 @@ def mutate(root, info, id):
-            figure.review_status = Figure.FIGURE_REVIEW_STATUS.REVIEW_IN_PROGRESS
+
+        figure.review_status = (
+            Figure.FIGURE_REVIEW_STATUS.REVIEW_IN_PROGRESS
+            if figure.figure_review_comments.all().count() > 0
+            else Figure.FIGURE_REVIEW_STATUS.REVIEW_NOT_STARTED
+        )
         figure.approved_by = None
         figure.approved_on = None
         figure.save()
 
-        if figure.event:
-            # Update event status
-            figure.update_event_status(figure.event)
-
-            _type = None
-            if figure.event and figure.event.review_status == Event.EVENT_REVIEW_STATUS.APPROVED:
-                _type = Notification.Type.FIGURE_UNAPPROVED_IN_APPROVED_EVENT
-            if figure.event and figure.event.review_status == Event.EVENT_REVIEW_STATUS.SIGNED_OFF:
-                _type = Notification.Type.FIGURE_UNAPPROVED_IN_SIGNED_EVENT
-            if _type:
-                Notification.send_multiple_notifications(
-                    recipients=figure.event.regional_coordinators(
-                        event=figure.event,
-                        figure=figure,
-                    ),
-                    type=_type,
-                    actor=info.context.user,
-                    event=figure.event,
-                    figure=figure,
-                )
-        # NOTE: Event is updated on figure save using signals.
-        # We are clearing figure and its reference objects (event).
-        figure.refresh_from_db()
+        def _get_notification_type(event):
+            if event.review_status == Event.EVENT_REVIEW_STATUS.APPROVED:
+                return Notification.Type.FIGURE_UNAPPROVED_IN_APPROVED_EVENT
+            if event.review_status == Event.EVENT_REVIEW_STATUS.SIGNED_OFF:
+                return Notification.Type.FIGURE_UNAPPROVED_IN_SIGNED_EVENT
+            return None
+
+        _type = _get_notification_type(figure.event)
```

Check if figure.event exists

---

We do not need to check if figure.event exits because it should always exist.

---

@tnagorra  event is nullable field

---

A figure should not have a non-nullable event. If there are some in the database it should crash instead of silently being ignore.